### DISCUSSION TITLE
Reconnect on network change

### DISCUSSION THREAD

**Discussion by stevemolloy** [profanity-im/profanity#1851] [Q&A]
Hi all,
My job requires me to frequently change network.  Wireless to wired, activating and deactivating VPN, waking the laptop in a different network from the one you put it to sleep in, etc.

Profanity doesn't seem to reconnect when this happens, and I wonder if you have any tips.  At the moment I'm dealing with it by quitting and restarting profanity, which isn't particularly smooth :)

Any advice would be much appreciated :)

**Comment by jubalh** [comment]
@sjaeckel probably has some knowledge on this

**Comment by H3rnand3zzz** [comment]
If you are using the latest build, you can use `/reconnect now` on disconnect (fixed thanks to @sjaeckel ). We are currently working on fix for this networking issue.

**Comment by H3rnand3zzz** [comment]
Please, use `/autoping` feature. It's the best solution at the moment. Bundled with `/reconnect` you can have it working just perfect.
Example of usage:
```
/autoping set 15
/autoping timeout 5
/reconnect 10
```

You can adjust numbers to your preference.

Also you can use `/reconnect now` to reconnect if you are using the latest build of the profanity.

As an alternative approach, you can use a hack. You can use profanity with Proxychains. TOR network is highly sensitive to networking problems and will disconnect immediately on problems and thus it can replace autoping solution, all that would be needed is to set up `/reconnect`. Please, note that TOR network might be blocked/raise red flags in corporate networks since it might be used for malicious activities.

Sorry for the late response, it required some in-depth investigation on networking problem. And feedback regarding usefulness if this advise would be highly appreciated :)

**Comment by H3rnand3zzz** [comment]
@stevemolloy, could you solve the issue with the way described above? If so, please, mark it as an answer so if other people have the same problem, they could find solution easier.
